native/mod_proxy_cluster/mod_proxy_cluster.c

```diff
@@ -2198,6 +2198,40 @@ static proxy_worker *internal_find_best_byrequests(proxy_balancer *balancer, pro
     return mycandidate;
 }
 
+/* Get the http worker if we are getting a STATUS for ws one. */
+static proxy_worker *get_http_worker(server_rec *s, proxy_worker *ws_worker)
+{
+    if (ws_worker->balancer != NULL) {
+        /* we need to find the corresponding http/https worker */
+        char *ptrw;
+        proxy_balancer *balancer = ws_worker->balancer;
+        int sizew = balancer->workers->elt_size;
+        int j;
+        ptrw = balancer->workers->elts;
+        for (j = 0; j < balancer->workers->nelts; j++,  ptrw=ptrw+sizew) {
+            proxy_worker **worker = (proxy_worker **) ptrw;
+            if (strcasecmp((*worker)->s->scheme, "HTTP") == 0 ||
+                strcasecmp((*worker)->s->scheme, "HTTPS") == 0) {
+                if ((*worker)->s->port == ws_worker->s->port &&
+                    !strcmp((*worker)->s->route, ws_worker->s->route) &&
+                    !strcmp((*worker)->s->hostname, ws_worker->s->hostname)) {
```

Comment:
It's a bit weird to me that for `strcasecmp` it is used `== 0` and for `strcmp` it's negation. I would unify that. The former seems a bit cleaner.

Reply:
There are more place where we have  == 0 in the conditions I think we should cleanup all those in a separate PR.